src/docs/product/issues/issue-details/performance-issues/db-main-thread-io.mdx

```diff
@@ -0,0 +1,32 @@
+---
+title: "Database on Main Thread"
+sidebar_order: 20
+description: "Learn more about Database on Main Thread and how to diagnose and fix these issues."
+---
+
+<Note>
+
+This feature is available **only for iOS devices** and if you're in the Early Adopter program. Features available to Early Adopters are still in-progress and may have bugs. We recognize the irony.
+
+If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily.
+
+</Note>
+
+
+The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread should not be used for performing long-running operations like database queries. These kinds of operations block the whole UI until they finish running and get in the way of the user interacting with the app.
+
```

Comment:
```suggestion
The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread shouldn't be used for performing long-running operations like database queries. These kinds of operations block the whole UI until they finish running and get in the way of the user interacting with the app.

```